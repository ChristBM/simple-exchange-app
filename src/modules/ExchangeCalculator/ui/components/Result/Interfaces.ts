import { ConversionType, ExchangeRateType } from '@exCalcComponents/Calculator/Interfaces';

export type ResultProps = {
  labelA?: string;
  labelB?: string;
  conversion?: ConversionType;
  exchangeRate?: ExchangeRateType;
}
