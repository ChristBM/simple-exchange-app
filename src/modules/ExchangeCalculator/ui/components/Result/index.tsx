import { IconArrowsExchange } from '@tabler/icons-react';

import { ResultProps } from './Interfaces';
import {
  ConversionReadOnly,
  ConversionResultWrapper,
  CurrencyText,
  ExchangeResultWrapper,
  ResultLabelText,
  ResultSectionWrapper,
  ResultWrapper,
} from './styles';

const RESULT_DEFAULT_PROPS = {
  labelA: 'Conversion Result:',
  labelB: 'Exchange Rate:',
  conversion: {
    result: 0,
    currency: '---',
  },
  exchangeRate: {
    fromCurrency: '---',
    result: 1,
    toCurrency: '---',
  },
};

export default function Result({
  labelA = RESULT_DEFAULT_PROPS.labelA,
  labelB = RESULT_DEFAULT_PROPS.labelB,
  conversion = RESULT_DEFAULT_PROPS.conversion,
  exchangeRate = RESULT_DEFAULT_PROPS.exchangeRate,
}: ResultProps) {
  return (
    <ResultWrapper>
      <ResultSectionWrapper>
        <ResultLabelText>{labelA}</ResultLabelText>

        <ConversionResultWrapper>
          <ConversionReadOnly type="text" readOnly value={conversion.result} />

          <CurrencyText>{conversion.currency}</CurrencyText>
        </ConversionResultWrapper>
      </ResultSectionWrapper>

      <ResultSectionWrapper>
        <ResultLabelText>{labelB}</ResultLabelText>

        <ExchangeResultWrapper>
          <p>1</p>
          <CurrencyText>{exchangeRate.fromCurrency}</CurrencyText>

          <div>
            <IconArrowsExchange
              stroke={2}
              color="purple"
              size={20}
              style={{ paddingBlockStart: '3px' }}
            />
          </div>

          <ConversionReadOnly type="text" readOnly value={exchangeRate.result} />

          <CurrencyText>{exchangeRate.toCurrency}</CurrencyText>
        </ExchangeResultWrapper>
      </ResultSectionWrapper>
    </ResultWrapper>
  );
}
