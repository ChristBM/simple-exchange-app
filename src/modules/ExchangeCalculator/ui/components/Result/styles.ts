'use client';

import styled from 'styled-components';

export const ResultWrapper = styled.div`
  display: flex;
  width: 100%;
  height: max-content;
  flex-direction: column;
  gap: 30px;
`;

export const ResultSectionWrapper = styled.section`
  display: flex;
  flex-flow: column wrap;
  border-bottom: 1px solid #373a40;
  align-items: baseline;
  gap: 10px;
`;

export const ResultLabelText = styled.p`
  font-size: 14px;
  letter-spacing: 1px;
  font-weight: 700;
`;

export const ConversionResultWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: baseline;
  font-size: 18px;
  gap: 5px;
`;

export const ConversionReadOnly = styled.input`
  appearance: none;
  width: 190px;
  background: none;
  border: none;
  outline: none;

  &::selection {
    background-color: none;
    color: cyan;
  }
`;

export const ExchangeResultWrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: baseline;
  font-size: 18px;
  gap: 5px;
`;

export const CurrencyText = styled.p`
  font-size: 14px;
  letter-spacing: 1px;
  font-weight: 300;
`;
