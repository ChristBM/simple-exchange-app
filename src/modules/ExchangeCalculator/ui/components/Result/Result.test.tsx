/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import { create } from 'react-test-renderer';
import Result from '.';

describe('<Result/> snapshot', () => {
  it('Renders correctly', () => {
    // Given + When
    const tree = create(<Result />).toJSON();

    // Then
    expect(tree).toMatchSnapshot();
  });
});
