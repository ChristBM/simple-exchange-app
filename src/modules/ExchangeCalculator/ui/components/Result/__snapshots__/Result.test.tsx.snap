// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result/> snapshot Renders correctly 1`] = `
<div
  className="sc-6a4bc0eb-0 kUTnqB"
>
  <section
    className="sc-6a4bc0eb-1 hGsTsy"
  >
    <p
      className="sc-6a4bc0eb-2 diBJjv"
    >
      Conversion Result:
    </p>
    <div
      className="sc-6a4bc0eb-3 dIZZvP"
    >
      <input
        className="sc-6a4bc0eb-4 dgxDd"
        readOnly={true}
        type="text"
        value={0}
      />
      <p
        className="sc-6a4bc0eb-6 kdDtIb"
      >
        ---
      </p>
    </div>
  </section>
  <section
    className="sc-6a4bc0eb-1 hGsTsy"
  >
    <p
      className="sc-6a4bc0eb-2 diBJjv"
    >
      Exchange Rate:
    </p>
    <div
      className="sc-6a4bc0eb-5 goOuKl"
    >
      <p>
        1
      </p>
      <p
        className="sc-6a4bc0eb-6 kdDtIb"
      >
        ---
      </p>
      <div>
        <svg
          className="tabler-icon tabler-icon-arrows-exchange"
          fill="none"
          height={20}
          stroke="purple"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            {
              "paddingBlockStart": "3px",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 10h14l-4 -4"
          />
          <path
            d="M17 14h-14l4 4"
          />
        </svg>
      </div>
      <input
        className="sc-6a4bc0eb-4 dgxDd"
        readOnly={true}
        type="text"
        value={1}
      />
      <p
        className="sc-6a4bc0eb-6 kdDtIb"
      >
        ---
      </p>
    </div>
  </section>
</div>
`;
